docs/build/wasm/from-zero-to-ink-hero/manic-minter/manic-e2e.md

```diff
@@ -0,0 +1,143 @@
+}
+```
+
+## Set ManicMinter as Owner of Oxygen
+We will use the `build_message` macro to compose the `transfer_ownership` method of the Oxygen contract. The `client.call()` will be used to execute the contract call. The `call_dry_run` method with `owner()` message will be used to verify the result of the contract call.
+
+```rust
+// Set ManicMinter contract to be the owner of Oxygen contract
+let change_owner = build_message::<OxygenRef>(oxygen_account_id.clone())
+    .call(|p| p.transfer_ownership(manic_minter_account_id));
+client
+    .call(&ink_e2e::alice(), change_owner, 0, None)
+    .await
+    .expect("calling `transfer_ownership` failed");
+
+// Verify that ManicMinter is the Oxygen contract owner
+let owner = build_message::<OxygenRef>(oxygen_account_id.clone()).call(|p| p.owner());
+let owner_result = client
+    .call_dry_run(&ink_e2e::alice(), &owner, 0, None)
+    .await
+    .return_value();
+assert_eq!(owner_result, manic_minter_account_id);
+```
+
+## Set Price for Oxygen Tokens
+
+We will use the `build_message` macro to compose the `set_price` method of the ManicMinter contract. The `client.call()` will be used to execute the contract call. 
+
+```rust
+// Contract owner sets price
+let price_message = build_message::<ManicMinterRef>(manic_minter_account_id.clone())
+    .call(|manicminter| manicminter.set_price(PRICE));
+client
+    .call(&ink_e2e::alice(), price_message, 0, None)
+    .await
+    .expect("calling `set_price` failed");
+
+```
+## Mint Oxygen Tokens
+We are now ready to execute `manic_mint` method of the ManicMinter contract. We will use the `build_message` macro to compose the `manic_mint` method of the ManicMinter contract. The `client.call()` will be used to execute the contract call. The `call_dry_run` method with `balance_of()` message will be used to verify the result of the contract call on the Oxygen contract.
+
+```rust
+// Bob mints AMOUNT of Oxygen tokens by calling ManicMinter contract
+let mint_message = build_message::<ManicMinterRef>(manic_minter_account_id.clone())
+    .call(|manicminter| manicminter.manic_mint(AMOUNT));
+client
+    .call(&ink_e2e::bob(), mint_message, PRICE * AMOUNT, None)
+    .await
+    .expect("calling `pink_mint` failed");
+
+// Verify that tokens were minted on Oxygen contract
+let bob_account_id = get_bob_account_id();
+let balance_message = build_message::<OxygenRef>(oxygen_account_id.clone())
+    .call(|p| p.balance_of(bob_account_id));
+let token_balance = client
+    .call_dry_run(&ink_e2e::bob(), &balance_message, 0, None)
+    .await
+    .return_value();
+assert_eq!(token_balance, AMOUNT);
+```
+
+## Execute e2e Test
+The e2e tests are invoking the node which is running on the local machine. 
+Before running the test we need to setup the environment variable `CONTRACT_NODE` to the executable local node. The node can be Swanky-node or any other node that implements pallet-contracts.
+```bash
+export CONTRACTS_NODE="YOUR_CONTRACTS_NODE_PATH"
```

Comment:
can you add an example as well ?
`export CONTRACTS_NODE="/home/p/Documents/astar/Astar/target/release/astar-collator"`

Reply:
done